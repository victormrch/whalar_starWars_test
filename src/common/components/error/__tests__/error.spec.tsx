import { ErrorComponent } from '../index';
import { render, screen } from '@testing-library/react';

describe('Error component spec', () => {
	it('should render correctly and create snapshot', () => {
		const { asFragment } = render(<ErrorComponent />);

		expect(asFragment()).toMatchSnapshot();
	});
	it('should display and image and title in component', () => {
		const title = `Sorry, an error has ocurred!. We are trying to fix it! Please try again later!`;
		const imageTitle = 'R2D2 and C3PO error';
		render(<ErrorComponent />);

		const elementTitle = screen.getByRole('heading', {
			name: title,
		});

		const elementImage = screen.getByRole('img', {
			name: imageTitle,
		});

		expect(elementTitle).not.toBeNull();
		expect(elementTitle).toBeInTheDocument();
		expect(elementTitle.tagName).toEqual('H1');
		expect(elementTitle.textContent).toBe(title);

		expect(elementImage).not.toBeNull();
		expect(elementImage).toBeInTheDocument();
		expect(elementImage.tagName).toEqual('IMG');
	});
});
