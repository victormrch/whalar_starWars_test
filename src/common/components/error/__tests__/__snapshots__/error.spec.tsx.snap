// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component spec should render correctly and create snapshot 1`] = `
<DocumentFragment>
  <img
    alt="R2D2 and C3PO error"
    class="sc-gsnTZi jRTjwM"
  />
  <h1
    class="sc-bczRLJ jylBfq"
  >
    Sorry, an error has ocurred!. We are trying to fix it! 
    <br />
    Please try again later!
  </h1>
</DocumentFragment>
`;
